Sources/ApodiniNetworking/URI.swift

```diff
@@ -0,0 +1,123 @@
+//
+// This source file is part of the Apodini open source project
+//
+// SPDX-FileCopyrightText: 2019-2021 Paul Schmiedmayer and the Apodini project authors (see CONTRIBUTORS.md) <paul.schmiedmayer@tum.de>
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import ApodiniUtils
+
+
+/// A URL
+public struct URI: LosslessStringConvertible {
```

Comment:
I intentionally called the type `URI` instead of `URL` to avoid naming ambiguities with Foundation's URL type. IIRC a URL and a URI are effectively the same thing; will update the comment though.

(The reason why this type exists in the first place it that Foundation's URL type implements a lot of unnecessary functionality we're not interested in, and also because the URL type defines all of its properties as optional, whereas we want things scheme, hostname, or path to be non-optional.)

Reply:
Sounds good 👍